clang/lib/Driver/ToolChains/Gnu.cpp

```diff
@@ -598,7 +598,8 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   // linked archives.  The unbundled information is a list of files and not
   // an actual object/archive.  Take that list and pass those to the linker
   // instead of the original object.
-  if (JA.isDeviceOffloading(Action::OFK_OpenMP)) {
+  if (JA.isDeviceOffloading(Action::OFK_OpenMP) ||
+      JA.isOffloading(Action::OFK_SYCL)) {
```

Comment:
Just wondering, why do we need this change now?

Reply:
There is now an incoming `TY_Tempfilelist` that contains the list of images unbundled from a given archive.  This never happened before for the SYCL offloading toolchain.  So open this up so we can add the `@` appropriately.